configure.in.in

```diff
@@ -106,9 +106,19 @@ OFF_T_FLAGS="-define:OFF_T_$ac_cv_sizeof_off_t"
 AC_SUBST(OFF_T_FLAGS)
 
 MONO_REQUIRED_VERSION=1.0
+FIRST_MONO_VERSION_WITH_NET_4_6_SUPPORT=4.4
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
 if test "x$has_mono" = "xfalse" ; then
 	PKG_CHECK_MODULES(MONO_DEPENDENCY, mono-2 >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
+	if test "x$has_mono" = "xtrue" ; then
```

Comment:
Beware, you placed this  `if test "x$has_mono" = "xtrue"` check inside a has_mono=xfalse block, so potentially this may always evaluate to false.